Examples/Io/Root/src/RootParticleWriter.cpp

```diff
@@ -105,18 +102,16 @@ ActsExamples::ProcessCode ActsExamples::RootParticleWriter::finalize() {
 
 ActsExamples::ProcessCode ActsExamples::RootParticleWriter::writeT(
     const AlgorithmContext& ctx, const SimParticleContainer& particles) {
+  const SimParticleContainer* finalParticles = nullptr;
+  if (m_inputFinalParticles.isInitialized()) {
+    finalParticles = &m_inputFinalParticles(ctx);
+  }
+
```

Comment:
Could we move this block down, next to
```
    bool wroteFinalParticle = false;
    if (finalParticles != nullptr) {
```
?

Reply:
the other section is in a loop and we usually query all the whiteboard entries at the beginning of the algorithm